ipn/local.go

```diff
@@ -1338,18 +1339,26 @@ func (b *LocalBackend) enterState(newState State) {
 
 	switch newState {
 	case NeedsLogin:
+		b.mu.Lock()
+		systemd.Status("Needs login: %s", b.authURL)
+		b.mu.Unlock()
 		b.blockEngineUpdates(true)
 		fallthrough
 	case Stopped:
 		err := b.e.Reconfig(&wgcfg.Config{}, &router.Config{})
 		if err != nil {
 			b.logf("Reconfig(down): %v", err)
 		}
+
+		if b.authURL == "" {
```

Comment:
Assumption: b.authURL is an empty string when there is not one set. This is used so that the needs login status isn't immediately eaten by the stopped case. I don't know if this assumption holds true.